### PR TITLE
Introduce an intermediate representation for road

### PR DESCRIPTION
**Pull request by Jackwgm**
The idea is that by storing the type of each road segment (turn, t-junction, etc) and it's orientation, in addition to just the coords, it should be easier to draw additional details on to the road, e.g. houses.

The RoadBuilder creates a list of RoadSegment objects from the loaded Level. Then the RoadDrawer draws the GameObjects from this.

### CODE REVIEW

**Comment by JacqMT** [comment]
Can this be sorted so that it only contains the relevant commits please? Looks like it's doing what mine did before and repeating the history from the beginning of the branch rather than the change history compared to master. A new branch should fix it, just then use 'git cherry-pick COMMIT_SHA' for the new ones you want to be reviewed

File: Assets/Scripts/RoadBuilder.cs [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+	{
+		public RoadType roadType;
+		public Coordinate coords;
+		public Direction direction;
+
+		public RoadSegment (RoadType roadType, Coordinate coords, Direction direction)
+		{
+			this.roadType = roadType;
+			this.coords = coords;
+			this.direction = direction;
```
Comment (mrniket) [inline Assets/Scripts/RoadBuilder.cs]: how does the direction work when it comes to a Turn road type?

File: Assets/Scripts/RoadBuilder.cs [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+				roadSegments [currentIndex] = CreateRoadSegment (nodes, currentIndex);
+			}
+			return roadSegments;
+		}
+
+		private RoadSegment CreateRoadSegment (PathNode[] nodes, int currentNodeIndex)
+		{
+			PathNode node = nodes [currentNodeIndex];
+			switch (node.connectedNodes.Length) {
+			case 1:
```
Comment (mrniket) [inline Assets/Scripts/RoadBuilder.cs]: you should be able to use the enum names here

**Comment by Jackwgm** [comment]
I've raised a different MR that doesn't have all the extra commits. I'll address the comments you've raised on that

**Pull request closed without merging**